src/Header/RequestParser.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OperateData.h"
+
+//used to make request string to OperateData(JSON like) format.
+//string inRequest "GET /?token=xxx HTTP/1.1/n..." => unique_ptr<OperateData> data = ParseHeader(inRequest)
+//data will be a json like format map, header and body will be seperate to two OperateData
+class RequestParser
+{
+public:
+    //singleton code convention method.
```

Comment:
If all methods are static, you don't need singleton. Simply use the static methods directly.

Reply:
Parse function changed to non static